app/models/lead.rb

```diff
@@ -231,9 +242,34 @@ def all_children_standard_key(node = self, result = [], depth = 0) # couplets be
     result
   end
 
+  # Return value always lists self first.
```

Comment:
There is a position-based paradigm/pattern for updating positions that might be cleaner.  In essence you POST to somethign like a `reorder` action the ids in order, in an array. THen you loop that array and simply update the position (without validation callbacks etc).  However, I'm not completely sure that meets the need in this method.